tests/filecheck/dialects/riscv_scf/loop_range_folding.mlir

```diff
@@ -1,47 +1,56 @@
 // RUN: xdsl-opt -p riscv-scf-loop-range-folding %s | filecheck %s
 
-%0, %1, %2, %3 = "test.op"() : () -> (!riscv.freg<>, !riscv.reg<>, !riscv.reg<>, !riscv.reg<>)
-%c0 = riscv.li 0 : () -> !riscv.reg<>
-%c64 = riscv.li 64 : () -> !riscv.reg<>
-%c1 = riscv.li 1 : () -> !riscv.reg<>
+// CHECK:       builtin.module {
 
-riscv_scf.for %arg4 : !riscv.reg<> = %c0 to %c64 step %c1 {
-    %4 = riscv.li 4 : () -> !riscv.reg<>
-    %5 = riscv.mul %arg4, %4 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
-    "test.op"(%5) : (!riscv.reg<>) -> ()
-    %6 = riscv.li 4 : () -> !riscv.reg<>
-    %7 = riscv.mul %arg4, %6 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
-    "test.op"(%7) : (!riscv.reg<>) -> ()
-}
+%lb, %ub, %step = "test.op"() : () -> (!riscv.reg<>, !riscv.reg<>, !riscv.reg<>)
+%0, %1 = "test.op"() : () -> (!riscv.reg<>, !riscv.reg<>)
+// CHECK-NEXT:    %{{.*}}, %{{.*}}, %{{.*}} = "test.op"() : () -> (!riscv.reg<>, !riscv.reg<>, !riscv.reg<>)
+// CHECK-NEXT:    %{{.*}}, %{{.*}} = "test.op"() : () -> (!riscv.reg<>, !riscv.reg<>)
 
-// Don't hoist multiplication by different constants
-riscv_scf.for %arg4 : !riscv.reg<> = %c0 to %c64 step %c1 {
-    %4 = riscv.li 4 : () -> !riscv.reg<>
-    %5 = riscv.mul %arg4, %4 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
-    %6 = riscv.li 5 : () -> !riscv.reg<>
-    %7 = riscv.mul %arg4, %6 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+riscv_scf.for %2 : !riscv.reg<> = %lb to %ub step %step {
+    // mul by constant
+    %3 = riscv.li 3 : () -> !riscv.reg<>
+    %4 = riscv.mul %2, %3 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+    // add with constant
+    %5 = riscv.li 5 : () -> !riscv.reg<>
+    %6 = riscv.add %4, %5 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+    "test.op"(%6) : (!riscv.reg<>) -> ()
 }
+// CHECK-NEXT:    %{{.*}} = riscv.li 3 : () -> !riscv.reg<>
+// CHECK-NEXT:    %{{.*}} = riscv.mul %{{.*}}, %{{.*}} : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+// CHECK-NEXT:    %{{.*}} = riscv.mul %{{.*}}, %{{.*}} : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+// CHECK-NEXT:    %{{.*}} = riscv.mul %{{.*}}, %{{.*}} : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+// CHECK-NEXT:    %{{.*}} = riscv.li 5 : () -> !riscv.reg<>
+// CHECK-NEXT:    %{{.*}} = riscv.add %{{.*}}, %{{.*}} : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+// CHECK-NEXT:    %{{.*}} = riscv.add %{{.*}}, %{{.*}} : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+// CHECK-NEXT:    riscv_scf.for %{{.*}} : !riscv.reg<> = %{{.*}} to %{{.*}} step %{{.*}} {
+// CHECK-NEXT:      %{{.*}} = riscv.li 3 : () -> !riscv.reg<>
+// CHECK-NEXT:      %{{.*}} = riscv.li 5 : () -> !riscv.reg<>
+// CHECK-NEXT:      "test.op"(%{{.*}}) : (!riscv.reg<>) -> ()
+// CHECK-NEXT:    }
 
+// Don't fold
+riscv_scf.for %2 : !riscv.reg<> = %lb to %ub step %step {
+    // Two uses mul
```

Comment:
can you use proper english for this comment please? i cannot parse whoever many times I read this